osu.Game.Rulesets.Taiko/Objects/DrumRoll.cs

```diff
@@ -106,7 +111,18 @@ private void createTicks(CancellationToken cancellationToken)
             }
         }
 
-        public override Judgement CreateJudgement() => new TaikoDrumRollJudgement();
+        public void SetBonus(bool bonus)
```

Comment:
Should be virtual maybe

Reply:
I don't see why this should be virtual, please clarify.

Reply:
Don't know if I'd like to add this function to the HitObject base class, as it's only use is in these 2 kinds of HitObjects, which only appear in taiko, will do so however if one of the maintainers think this is a good idea because of the casts.

Reply:
> Don't know if I'd like to add this function to the HitObject base class, as it's only use is in these 2 kinds of HitObjects, which only appear in taiko, will do so however if one of the maintainers think this is a good idea because of the casts. To that, this kind of code is wayy more verbose than, say, this
> 
> ```cs
>  public void ApplyToHitObject(HitObject hitObject) 
>  { 
>      hitObject.SetBonus(true);
>  } 
> ```
> 
> While this is simpler, it does a worse job explaining itself.

Not `HitObject` class but `TaikoHitObject` makes sense

Reply:
Applying it to the `TaikoHitObject` however would still require a cast in the `ApplyToHitObject` function, as it takes a generic `HitObject` and is supplied by the mod interface, so it wouldn't make the function much simpler.